pkg/server/options/authentication.go

```diff
@@ -58,7 +59,7 @@ func (s *AdminAuthentication) Validate() []error {
 	errs := []error{}
 
 	if s.TokenHashFilePath == "" && s.KubeConfigPath != "" {
-		errs = append(errs, fmt.Errorf("--admin-kubeconfig requires --admin-token-hash-file-path"))
+		errs = append(errs, fmt.Errorf("--kubeconfig-path requires --admin-token-hash-file-path"))
```

Comment:
am not a fan of this change. Just add another flag for the other kubeconfig.

Reply:
This line is actually correcting a mismatch in option names.  `--admin-kubeconfig` is not an actual flag.  Given that, I know I am still changing the meaning of `--kubeconfig-path`.  Would you still like me to add another flag for the new kubeconfig?

Reply:
ic.

But still wondering what we actually want here. Why again do we need the same kubeconfig with external and internal address? Why not always the external one?

We had a discussion about a second kubeconfig for other components, with only one context. This PR is not that, is it?

Reply:
Wrote up [a comment on the linked issue](https://github.com/kcp-dev/kcp/issues/707#issuecomment-1073411354) to try and explain my thoughts as I worked on this PR.